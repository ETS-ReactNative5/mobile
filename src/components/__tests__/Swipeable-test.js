import 'react-native'
import React from 'react'
import renderer from 'react-test-renderer'
jest.mock('../SwipeSubject', () => 'SwipeSubject')

import { Swipeable } from '../Swipeable'

const workflow = {
  first_task: 'T0',
  tasks: {
    T0: {
      question: 'Is it?',
      answers: [ { label: 'Yes' }, { label: 'No' } ],
    }
  }
}
const subjectSizes = {
  resizedWidth: 100,
  resizedHeight: 100
}
const subject = {
    id: '3424'
}
const nextSubject = {
    id: '234'
}
it('renders correctly', () => {
  const tree = renderer.create(
    <Swipeable
      saveAnnotation={jest.fn}
      saveThenStartNewClassification={jest.fn}
      workflow={workflow}
      subjectSizes={subjectSizes}
      subject={subject}
      nextSubject={nextSubject}
      />
  ).toJSON()
  expect(tree).toMatchSnapshot()
})
