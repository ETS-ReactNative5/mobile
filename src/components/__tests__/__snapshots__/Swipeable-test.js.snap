// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Array [
      undefined,
      Object {
        "height": 140,
        "top": NaN,
      },
    ]
  }
>
  <View
    style={undefined}
  >
    <View
      collapsable={undefined}
      onMoveShouldSetResponder={[Function]}
      onMoveShouldSetResponderCapture={[Function]}
      onResponderEnd={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderReject={[Function]}
      onResponderRelease={[Function]}
      onResponderStart={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      onStartShouldSetResponderCapture={[Function]}
      style={
        Object {
          "height": 110,
          "transform": Array [
            Object {
              "translateX": 0,
            },
            Object {
              "translateY": 0,
            },
            Object {
              "rotate": "0deg",
            },
          ],
          "width": 100,
        }
      }
    >
      <View
        accessibilityComponentType={undefined}
        accessibilityLabel={undefined}
        accessibilityTraits={undefined}
        accessible={true}
        collapsable={undefined}
        hasTVPreferredFocus={undefined}
        hitSlop={undefined}
        isTVSelectable={true}
        nativeID={undefined}
        onLayout={undefined}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "height": 110,
            "opacity": 1,
            "width": 100,
          }
        }
        testID={undefined}
        tvParallaxProperties={undefined}
      >
        <SwipeSubject
          inFront={true}
          leftAnswer="No"
          pan={
            Object {
              "x": 0,
              "y": 0,
            }
          }
          rightAnswer="Yes"
          seenThisSession={undefined}
          setNextSubject={undefined}
          subject={
            Object {
              "id": "3424",
            }
          }
          subjectSizes={
            Object {
              "resizedHeight": 100,
              "resizedWidth": 100,
            }
          }
        />
      </View>
    </View>
  </View>
</View>
`;
